File: backend/app/models/agent.py
from sqlalchemy import JSON, String, Text
from sqlalchemy.orm import Mapped, mapped_column

from app.models.base import Base, TimestampMixin


class Agent(TimestampMixin, Base):
    __tablename__ = "agents"
    name: Mapped[str] = mapped_column(String(200), nullable=False, unique=True)
    description: Mapped[str | None] = mapped_column(Text, nullable=True)
    capability_tags: Mapped[dict | None] = mapped_column(JSON, default=list)
    dify_endpoint: Mapped[str] = mapped_column(String(500), nullable=False)
    dify_api_key: Mapped[str] = mapped_column(String(500), nullable=False)
    input_params: Mapped[dict] = mapped_column(JSON, nullable=False)
    output_params: Mapped[dict] = mapped_column(JSON, nullable=False)
    timeout_seconds: Mapped[int] = mapped_column(default=300)
    auto_execute: Mapped[bool] = mapped_column(default=False)
    confirm_before_exec: Mapped[bool] = mapped_column(default=True)
    is_enabled: Mapped[bool] = mapped_column(default=True)
    call_count: Mapped[int] = mapped_column(default=0)
    success_count: Mapped[int] = mapped_column(default=0)

File: frontend/src/types/agent.ts
import type { ParamDefinition } from './api';

export interface Agent {
  id: string;
  name: string;
  description?: string;
  capability_tags: string[];
  dify_endpoint: string;
  dify_api_key: string;
  input_params: ParamDefinition[];
  output_params: ParamDefinition[];
  timeout_seconds: number;
  auto_execute: boolean;
  confirm_before_exec: boolean;
  is_enabled: boolean;
  call_count: number;
  success_count: number;
  created_at: string;
  updated_at: string;
  user_id: string;
}
